### PR TITLE
add test labels

### PR DESCRIPTION
**Pull request by cmcmarrow**
### What does this PR do?
Improves salt testing

### What issues does this PR fix or reference?
Fixes: Allows contributors to get test results fast https://github.com/saltstack/salt/issues/63081

### Previous Behavior
* A PR would run fast test 2 times per run.

### New Behavior
* A PR does not run fast tests 2 times per run.
* People can add labels to change what tests run.

##
new test decorators `--no-fast-tests`, `--slow-tests`,  `--core-tests`, `--flaky-jail`
* `--fast-test`, default = true
Salt contributors never need to write this pytest decorator. A test will default to this decorator if the test does not have `speed_test`, `slow_test` and `core_test`. Most fast-test should take less then 5 secs to run.
* `--slow-tests` or `--run_slow`, default = false
A test that takes more than 5 secs to run.
 *  `--core-tests`, default = false
A test of a low level salt function that is rarely if ever is touched by contributors.
 * `--flaky-jail`, default = false
```
#EX
#@pytest.mark.core_test
#@pytest.mark.flaky_jail
@pytest.mark.slow_test
def test():
   ...
```
##
### Merge requirements satisfied?
- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -175,6 +175,40 @@ def pytest_addoption(parser):
...
+        "--core-tests",
+        dest="core",
+        action="store_true",
+        default=False,
+        help=(
+            "Run salt-core tests. These tests test the engine of salt! "
+            "Default: False"
+        ),
+    )
+    test_selection_group.addoption(
```
Comment (Ch3LL) [inline tests/conftest.py:189]: Why do we need a fast group, when we already have a slow group. The tests should run the fast tests automatically and then also run the slow tests if `--run-slow` is specified.

**Review by Ch3LL — COMMENTED** [review]
Just a question

File: tests/conftest.py (hunk 1 above)
Comment (cmcmarrow) [reply tests/conftest.py:189]: it so if you want to exclude fast tests. For examples your fast test all pass, but you want to test your slow test. You dont want to wait for another 15 mins for the fast tests to run again.

Comment (cmcmarrow) [reply tests/conftest.py:189]: core test are slow test but that also just test the low level stuff of salt like pillar or beacon

Comment (Ch3LL) [reply tests/conftest.py:189]: Do we have to decorate every single test that is not a slow or core test?

Comment (cmcmarrow) [reply tests/conftest.py:205]: no, pytest checks if !core_test or !slow_test -> treat test as fast_test

**Comment by cmcmarrow** [comment]
re-running all

File: changelog/63081.added.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Fix running fast tests 2x and add git labels to sweet
```
Comment (twangboy) [inline changelog/63081.added.md:1]: I don't understand what sweet means... maybe suite?

**Review by twangboy — CHANGES_REQUESTED** [review]

**Review by twangboy — APPROVED** [review]

**Pull request closed without merging**